### PR TITLE
Feature/link scraper as an action

### PR DESCRIPTION
**Pull request by Eyobyb**
# Description
- Add a link scraper action which will scrape and summarize with at most 3000 tokens of result. 
- Add a test for the link scraper actions 
# Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Maintenance
- [ ] New release

# Related issues
> Mention related GitHub and Linear issues. E.g. Closes #xxx or Fixes #xxx. Otherwise delete this section.

# Checklists
To speed up the review process, please follow these checklists:

## Development
- [x] The Pull Request is small and focused on one topic
- [x] Lint rules pass locally (`make format && make lint`)
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development (`make test`)
- [ ] The changes generate no new warnings (or explain any new warnings and why they're ok)
- [x] Commit messages are detailed
- [ ] Changed code is self-explanatory and/or I added comments
- [ ] I updated the documentation (docstrings, /docs)
See the testing guidelines for help on tests, especially those involving web services.

## Code review
- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached.
- [ ] I have performed a self-review of my code
- [ ] Issue from task tracker has a link to this pull request

💔 Thank you for submitting a pull request!

### CODE REVIEW

File: src/sherpa_ai/tools.py [hunk 1]
```diff
@@ -297,3 +305,80 @@ def _run(self, query: str) -> str:
...
+                    if git_scraper:
+                        scraped_data = {
+                            "data": git_scraper,
+                            "status": 200,
+                        }
+                    else:
+                        scraped_data = {"data": "", "status": 404}
+                else:
+                    scraped_data = scrape_with_url(link)
+                    print("%m" * 70)
```
Comment (saminegash) [inline src/sherpa_ai/tools.py:343]: remove print

File: src/sherpa_ai/tools.py [hunk 2]
```diff
@@ -297,3 +305,80 @@ def _run(self, query: str) -> str:
...
+                            link=link,
+                            question=query,
+                            text_data=chunk_summary,
+                            # user_id="",
+                            llm=self.llm,
+                        )
+
+                    final_summary.append({"data": chunk_summary, "link": link})
+                else:
+                    print("Scraping failed")
```
Comment (saminegash) [inline src/sherpa_ai/tools.py:370]: user logger.debug

File: src/tests/data/test_link_scraper_action_t_0.jsonl [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
...e the task mentioned. Include any links you used from the context and history in the result.\\nTask: what is unique about Ethiopia calendar? based on these links https://www.ethiopiancalendar.net/\\nResult:\\n", "agent": "human"}], "output": "The Ethiopian calendar is unique because it follows a different system than the Gregorian calendar commonly used in the Western world. The Ethiopian calendar has 13 months, with 12 months having 30 days each and a 13th month called Pagum\u0113 with 5 or 6 days, depending on whether it is a leap year. This makes the Ethiopian year shorter than the Gregorian year by about 7-8 years.\\n\\nAdditionally, the Ethiopian calendar is about 7-8 years behind the Gregorian calendar. For example, while the year 2021 is recognized in the Gregorian calendar, it is currently the year 2013 in the Ethiopian calendar.\\n\\nThe Ethiopian calendar also has its own unique New Year celebration called Enkutatash, which falls on September 11th (or September 12th in leap years) according to the Gregorian calendar.\\n\\nFor more information, you can visit the official website of the Ethiopian calendar at https://www.ethiopiancalendar.net/.", "llm_name": "gpt-3.5-turbo"}
```
Comment (saminegash) [inline src/tests/data/test_link_scraper_action_t_0.jsonl:4]: What is this file is it requered?

**Review by saminegash — CHANGES_REQUESTED** [review]

File: src/tests/data/test_link_scraper_action_t_0.jsonl (hunk 3 above)
Comment (Eyobyb) [reply src/tests/data/test_link_scraper_action_t_0.jsonl:4]: yes, Its auto generated

**Review by saminegash — APPROVED** [review]

**Pull request merged**